pydantic_ai_slim/pydantic_ai/_agent_graph.py

```diff
@@ -634,6 +640,24 @@ async def _handle_text_response(
             return self._handle_final_result(ctx, result.FinalResult(result_data), [])
 
 
+def cost(response: _messages.ModelResponse | models.StreamedResponse, ignore_warning_cost: bool) -> Decimal:
+    # If we can't calculate the price, we don't want to fail the run.
+    try:
+        cost = response.cost().total_price
+    except LookupError:
+        # NOTE(Marcelo): We can allow some kind of hook on the provider level, which we could retrieve via
+        # `ctx.deps.model.provider.calculate_cost`, but I'm not sure how would the API look like. Maybe a new parameter
+        # on the `Provider` classes, that parameter would be a callable that receives the same parameters as `genai_prices`.
+        if response.model_name not in ('test', 'function') and not ignore_warning_cost:
```

Comment:
I was thinking that test/function provider names would somehow make this special casing easier, I didn't even think about the model names. Since we can just use this, maybe the provider names aren't worth it. I especially don't like how the examples and docs now show `model_name='gpt-4o',` but also `provider_name='function',`